components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/internal/ApplicationManagementServiceComponentHolder.java

```diff
@@ -17,14 +17,19 @@
  */
 package org.wso2.carbon.identity.application.mgt.internal;
 
+import org.wso2.carbon.identity.application.mgt.AbstractInboundAuthenticatorConfig;
 import org.wso2.carbon.registry.api.RegistryService;
 import org.wso2.carbon.user.core.service.RealmService;
 import org.wso2.carbon.utils.ConfigurationContextService;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public class ApplicationManagementServiceComponentHolder {
 
     private static ApplicationManagementServiceComponentHolder instance=new
             ApplicationManagementServiceComponentHolder();
+    private static Map<String, AbstractInboundAuthenticatorConfig> inboundAuthenticatorConfigs = new HashMap<String, AbstractInboundAuthenticatorConfig>();
```

Comment:
We need to get the authenticator for the given type, there is a method called getInboundAuthenticatorConfig in the ApplicationManagementServiceComponentHolder which pass type and get the authenticator.